source/fundamentals/csfle.txt

```diff
@@ -0,0 +1,283 @@
+
+.. code-block:: java
+
+   import com.mongodb.ClientEncryptionSettings;
+   import com.mongodb.ConnectionString;
+   import com.mongodb.client.model.vault.DataKeyOptions;
+   import com.mongodb.client.vault.ClientEncryption;
+   import com.mongodb.client.vault.ClientEncryptions;
+   import org.bson.BsonBinary;
+   import org.bson.BsonDocument;
+
+   import java.util.Base64;
+
+   ...
+
+   String keyVaultNamespace = "admin.datakeys";
+   ClientEncryptionSettings clientEncryptionSettings = ClientEncryptionSettings.builder()
+           .keyVaultMongoClientSettings(MongoClientSettings.builder()
+                   .applyConnectionString(new ConnectionString("mongodb://localhost"))
+                   .build())
+           .keyVaultNamespace(keyVaultNamespace)
+           .kmsProviders(kmsProviders)
+           .build();
+
+   ClientEncryption clientEncryption = ClientEncryptions.create(clientEncryptionSettings);
+   BsonBinary dataKeyId = clientEncryption.createDataKey("local", new DataKeyOptions());
+   final String base64DataKeyId = Base64.getEncoder().encodeToString(dataKeyId.getData());
+
+   final String dbName = "test";
+   final String collName = "coll";
+   AutoEncryptionSettings autoEncryptionSettings = AutoEncryptionSettings.builder()
+           .keyVaultNamespace(keyVaultNamespace)
+           .kmsProviders(kmsProviders)
+           .schemaMap(new HashMap<String, BsonDocument>() {{
+               put(dbName + "." + collName,
+                       // Need a schema that references the new data key
+                       BsonDocument.parse("{"
+                               + "  properties: {"
+                               + "    encryptedField: {"
+                               + "      encrypt: {"
+                               + "        keyId: [{"
+                               + "          \"$binary\": {"
+                               + "            \"base64\": \"" + base64DataKeyId + "\","
+                               + "            \"subType\": \"04\""
+                               + "          }"
+                               + "        }],"
+                               + "        bsonType: \"string\","
+                               + "        algorithm: \"AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic\""
+                               + "      }"
+                               + "    }"
+                               + "  },"
+                               + "  \"bsonType\": \"object\""
+                               + "}"));
+           }}).build();
+
+Explicit Encryption and Decryption
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The prior examples demonstrated the automatic CSFLE feature. If your version of MongoDB does not offer automatic CSFLE,
+you can perform CSFLE explicitly. This method does not use the ``mongocryptd`` daemon / process. The
+``ClientEncryption`` class provides the mechanisms to use explicit encryption.
```

Comment:
I try to avoid "provides" since it might not be specific enough. How about:

```suggestion
``ClientEncryption`` class contains methods you can use to perform explicit encryption.
```